core/network/src/service.rs

```diff
@@ -361,82 +471,111 @@ fn start_thread<B: BlockT + 'static, S: NetworkSpecialization<B>, H: ExHashT>(
+			},
+			NetworkMsg::ReportPeer(who, severity) => {
+				match severity {
+					Severity::Bad(_) => network_service_2.lock().ban_node(who),
+					Severity::Useless(_) => network_service_2.lock().drop_node(who),
+					Severity::Timeout => network_service_2.lock().drop_node(who),
+				}
+			},
+			NetworkMsg::GetPeerId(who, sender) => {
+				let node_id = network_service_2
+					.lock()
+					.peer_id_of_node(who)
+					.cloned()
+					.map(|id| id.to_base58());
+				let _ = sender.send(node_id);
+			},
+		}
+		Ok(())
+	})
+	.then(|res| {
+		match res {
+			Ok(()) => (),
+			Err(_) => error!("Protocol disconnected"),
+		};
+		Ok(())
+	});
 
 	// The network service produces events about what happens on the network. Let's process them.
-	let network_service2 = network_service.clone();
-	let network = stream::poll_fn(move || network_service2.lock().poll()).for_each(move |event| {
-		let mut net_sync = NetSyncIo::new(&network_service, protocol_id);
-
+	let network = stream::poll_fn(move || network_service.lock().poll()).for_each(move |event| {
 		match event {
-			NetworkServiceEvent::ClosedCustomProtocols { node_index, protocols } => {
+			NetworkServiceEvent::ClosedCustomProtocols { node_index, protocols, debug_info } => {
 				if !protocols.is_empty() {
 					debug_assert_eq!(protocols, &[protocol_id]);
-					protocol.on_peer_disconnected(&mut net_sync, node_index);
+					let _ = protocol_sender.send(
+						ProtocolMsg::PeerDisconnected(node_index, debug_info));
 				}
 			}
-			NetworkServiceEvent::OpenedCustomProtocol { node_index, version, .. } => {
+			NetworkServiceEvent::OpenedCustomProtocol { node_index, version, debug_info, .. } => {
 				debug_assert_eq!(version, protocol::CURRENT_VERSION as u8);
-				protocol.on_peer_connected(&mut net_sync, node_index);
+				let _ = protocol_sender.send(ProtocolMsg::PeerConnected(node_index, debug_info));
 			}
-			NetworkServiceEvent::ClosedCustomProtocol { node_index, .. } => {
-				protocol.on_peer_disconnected(&mut net_sync, node_index);
+			NetworkServiceEvent::ClosedCustomProtocol { node_index, debug_info, .. } => {
+				let _ = protocol_sender.send(ProtocolMsg::PeerDisconnected(node_index, debug_info));
 			}
 			NetworkServiceEvent::CustomMessage { node_index, data, .. } => {
-				protocol.handle_packet(&mut net_sync, node_index, &data);
+				if let Some(m) = Decode::decode(&mut (&data as &[u8])) {
+					let _ = protocol_sender.send(ProtocolMsg::CustomMessage(node_index, m));
+					return Ok(())
+				}
+				let _ = network_sender.send(
+					NetworkMsg::ReportPeer(
+						node_index,
+						Severity::Bad("Peer sent us a packet with invalid format".to_string())
+					)
+				);
 			}
 			NetworkServiceEvent::Clogged { node_index, messages, .. } => {
-				protocol.on_clogged_peer(&mut net_sync, node_index,
-					messages.iter().map(|d| d.as_ref()));
+				debug!(target: "sync", "{} clogging messages:", messages.len());
+				for msg_bytes in messages.iter().take(5) {
```

Comment:
why take 5?

Reply:
This code is from master at https://github.com/paritytech/substrate/blob/d0f824f778faca2a2eda96c9cde2bf9cd888dbed/core/network/src/protocol.rs#L436

@tomaka would know the reason for it...

Reply:
I didn't put this. I imagine someone did because it was spamming the logs too much. It's a bit stupid as the whole point of dumping all the messages was to see the frequency's of each one.